.github/workflows/dev-publishing.yml

```diff
@@ -0,0 +1,57 @@
+name: Dev branch publishing
+
+on:
+  push:
+    branches:
+      - dev
+
+jobs:
+  dev_branch_publishing:
+    runs-on: ubuntu-latest
+
+    environment: npm-publish
+
+    if: github.repository == 'romcal/romcal'
+
+    env:
+      CI: true
+      GITHUB_USER: romcalproject
+      GITHUB_EMAIL: romcalproject@gmail.com
+
+    steps:
+      - name: Git checkout dev branch
+        uses: actions/checkout@v2
+        with:
+          ref: dev
+
+      - name: Install node.js and npm
+        uses: volta-cli/action@v1
+
+      - name: Install npm depencancies
+        run: npm ci --no-audit
```

Comment:
It doesn't matter, since `package-lock.json` isn't embedded in the NPM package. However you are right, it would be more logical to bump the version before running an install. I'll change the step orders.

Reply:
IMO it does matter, but not from the NPM package point of view.

1. We use `npm ci` which [by default uses `package-lock.json` if it exists](https://docs.npmjs.com/cli/v8/commands/npm-ci).
2. As I tried to demonstrate above, we need to _recreate_ `package-json.lock` after version bump and before commiting it. The easiest way is to run `npm i` (or `npm ci` but with `package-lock.json` removed); I tried to manually bump the version in `package-lock.json`, but `package-lock.json` simply gets modified whenever I install the deps in a clean (just cloned) repo. It does not happen when I recreate `package-lock.json` using `npm i`.

Reply:
And it might be reasonale to push the commit _after_ successful build and NPM release. What it the build/release for some reason fails?

Reply:
You are right, even if the `package-lock.json` isn't part of the NPM package, it must be updated in the codebase.
However, `phips28/gh-action-bump-version@master` is already taking care of updating it. So we are good 👍 .

And you also pointed out a good point in your last comment: the automated bump should be run just after the test step (to not create a new commit if the build or the tests are failing). I didn't see available options to bump the version without committing, but it seems reasonable to make the bump step just before the publishing step.
The publishing step can fail if the exact same npm version is already published, or if the NPM token isn't correctly set (so the risks are very limited from my point of view).

Reply:
> However, `phips28/gh-action-bump-version@master` is already taking care of updating it. So we are good :+1:.

I didn’t say it does not care of it (I have never ever used that package nor did I test it), but you said you tested this PR prior to sharing it with us and I have found out that the versions in `package.json` and `package-lock.json` differ, therefore I suspect it does not work as expected, but I might be wrong. :man_shrugging:

> the automated bump should be run just after the test step

I think that bumping should be done prior to installing dependences, but _only in `package*.json` files (and potentionally other files), but no commit should be created _until romcal is tested and packages released_.

As I said, I don’t use _any_ package/program to bump versions in my repos. I bump the version manually in `package.json` files, run `npm i`, test, release and if everything was successful, create a new commit. But then, the published packages might not point to the right commit, but I might be wrong.

>  The publishing step can fail if the exact same npm version is already published, or if the NPM token isn't correctly set (so the risks are very limited from my point of view).

I didn’t say the risks are high, but I personally like to be on the safer side if possible. But then, I leave the final decision up to you. :wink:

Reply:
Tell me if you think this PR is ready to be merged. I don’t see other issues with this PR.

Reply:
> therefore I suspect it does not work as expected

Ok, I understand now, thanks for your patience 😄. In fact, in this PR I've manually updated the version in the `package.json` file, but not in `package-lock.json`. It's now fixed.

Also, I've added a [missing feature](https://github.com/phips28/gh-action-bump-version/pull/139) in the `phips28/gh-action-bump-version` to prevent doing a commit just after the bump. Commit and tagging is now done just before publishing. So if the build or the tests are failing, no commit/tag is pushed to the repo, and the CI workflow is aborted.

If you don't see anything else, I think this PR is ready to be merged.